### PR TITLE
Use consistent data size units

### PR DESCRIPTION
**Pull request by carlcsaposs**
Signed-off-by: Carl Csaposs <carl@csaposs.com>

### CODE REVIEW

**Comment by carlcsaposs** [comment]
Should binary byte units be specified according to [ISO/IEC 80000-13](https://en.wikipedia.org/wiki/ISO/IEC_80000#Part_13:_Information_science_and_technology) (e.g. kibibyte, KiB)?

PHP uses "kilobyte" to represent 1024 bytes: https://www.php.net/manual/en/faq.using.php#faq.using.shorthandbytes
Nextcloud's front-end appears to do the same

File: developer_manual/client_apis/WebDAV/search.rst [hunk 1]
```diff
@@ -166,7 +166,7 @@ Get a file by id.
         </d:basicsearch>
     </d:searchrequest>
 
-Get all png and jpg files over 10MB.
+Get all png and jpg files over 10 MB.
```
Comment (carlcsaposs) [inline developer_manual/client_apis/WebDAV/search.rst:169]: This example uses 10 MB = 10000000 bytes (see line 207)

File: admin_manual/configuration_files/big_file_upload_configuration.rst [hunk 2]
```diff
@@ -1,14 +1,14 @@
 ===========================
-Uploading big files > 512MB
+Uploading big files > 512 MB
 ===========================
```
Comment (nickvergessen) [inline admin_manual/configuration_files/big_file_upload_configuration.rst:3]:
```suggestion
============================
```

Comment (nickvergessen) [inline admin_manual/configuration_files/big_file_upload_configuration.rst:3]:
```rst
============================
Uploading big files > 512 MB
============================
```

Need to adjust the count of `=` as well

File: admin_manual/configuration_files/big_file_upload_configuration.rst [hunk 3]
```diff
@@ -151,9 +151,9 @@ For upload performance improvements in environments with high upload bandwidth,
 
  sudo -u www-data php occ config:app:set files max_chunk_size --value 20971520
 
-Put in a value in bytes (in this example, 20MB). Set ``--value 0`` for no chunking at all.
+Put in a value in bytes (in this example, 20 MB). Set ``--value 0`` for no chunking at all.
 
-Default is 10485760 (10 MiB).
+Default is 10485760 (10 MB).
```
Comment (nickvergessen) [inline admin_manual/configuration_files/big_file_upload_configuration.rst:156]: MiB is actually correct here.
1MB = 1.000 KB
1 MiB = 1.024 KiB

File: admin_manual/configuration_server/config_sample_php_parameters.rst [hunk 4]
```diff
@@ -1909,7 +1909,7 @@ Values that undershoot the minimum will be ignored in favor of the minimum.
 
 	'hashingMemoryCost' => PASSWORD_ARGON2_DEFAULT_MEMORY_COST,
 
-The memory in KiB to be used by the algorithm for computing a hash. The value
+The memory in KB to be used by the algorithm for computing a hash. The value
```
Comment (nickvergessen) [inline admin_manual/configuration_server/config_sample_php_parameters.rst:1912]: also incorrect

File: admin_manual/installation/nginx.rst [hunk 5]
```diff
@@ -91,11 +91,11 @@ or CSS (.css) files via gzip could also cause such issues.
 Another cause of this issue could be not properly including mimetypes in the
 http block, as shown `here. <https://www.nginx.com/resources/wiki/start/topics/examples/full/>`_
 
-Upload of files greater than 10 MiB fails
+Upload of files greater than 10 MB fails
```
Comment (nickvergessen) [inline admin_manual/installation/nginx.rst:94]: also wrong

File: admin_manual/installation/nginx.rst [hunk 6]
```diff
@@ -91,11 +91,11 @@ or CSS (.css) files via gzip could also cause such issues.
...
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 If you configure nginx (globally) to block all requests to (hidden) dot files,
-it may be not possible to upload files greater than 10 MiB using the webpage
+it may be not possible to upload files greater than 10 MB using the webpage
```
Comment (nickvergessen) [inline admin_manual/installation/nginx.rst:98]: also wrong

**Review by nickvergessen — CHANGES_REQUESTED** [review]

**Comment by carlcsaposs** [comment]
> Should binary byte units be specified according to [ISO/IEC 80000-13](https://en.wikipedia.org/wiki/ISO/IEC_80000#Part_13:_Information_science_and_technology) (e.g. kibibyte, KiB)?
> 
> PHP uses "kilobyte" to represent 1024 bytes: https://www.php.net/manual/en/faq.using.php#faq.using.shorthandbytes Nextcloud's front-end appears to do the same

@nickvergessen Here's a branch with ISO 80000-13 units: https://github.com/nextcloud/documentation/compare/master...carlcsaposs:units-iso

Should the documentation follow the ISO 80000-13 standard or the [Nextcloud front-end](https://github.com/nextcloud/server/blob/6da8a6d62adde9b1ed011a7a18d43984cf5a27a8/lib/private/legacy/OC_Helper.php#L64) and PHP standard?

Please let me know if you'd like me to push the changes on the `carlcsaposs:units-iso` branch to this PR.

---

Also, it appears that the comments in Nextcloud's source code use inconsistent units (see [this example](https://github.com/nextcloud/server/blob/6da8a6d62adde9b1ed011a7a18d43984cf5a27a8/lib/private/legacy/OC_Files.php#L57) and [this example](https://github.com/nextcloud/server/blob/6da8a6d62adde9b1ed011a7a18d43984cf5a27a8/apps/files_external/3rdparty/icewind/smb/src/Native/NativeReadStream.php#L16)).

I'd be happy to submit PRs to update the units across Nextcloud's repositories--please let me know if that would be useful.

**Comment by tflidd** [comment]
It's better to call things what they are. Advantage of the Si-units is that you have MiB and MB to make a difference.

Question now would be, if php consiers 1M to be 1 MiB, and now you want to set a quota, do we show that you need to select a value in MiB, or do we ask a value in MB and then do the internal conversion to MiB to apply the limit via php?
And we have the clients as well.

There was a long discussion before the fork about the use of SI units: https://github.com/owncloud/core/issues/13386

**Comment by carlcsaposs** [comment]
@tflidd I appreciate you sharing the context of the owncloud issue--I didn't think to search owncloud repositories.  Thank you!

**Pull request closed without merging**